src/Blorc.OpenIdConnect/wwwroot/oidc-client-interop.js

```diff
@@ -27,15 +27,49 @@
 
                 let self = this;
 
-                if (config.timeForAutomaticLogoutOnUserInactivity > 0) {
-                    var inactivityTimer = setTimeout(trySignoutRedirect, config.timeForAutomaticLogoutOnUserInactivity);
+                if (config.timeForUserInactivityAutomaticLogout > 0) {
+                    var userInactivityLogoutTimer = setTimeout(trySignoutRedirect, config.timeForUserInactivityAutomaticLogout);
 
-                    document.onmousemove = resetInactivityTimer;
-                    document.onkeypress = resetInactivityTimer;
+                    var userInactivityTimer;
+                    if (config.timeForUserInactivityNotification > 0) {
+                        userInactivityTimer = setTimeout(notifyUserInactivity, config.timeForUserInactivityNotification);
+                    }
+
+                    document.addEventListener('mousemove', function (event) { resetInactivityTimers(); });
+                    document.addEventListener('keypress', function (event) { resetInactivityTimers(); });
+
+                    var notifiedByUserInactivity = false;
+
+                    function resetInactivityTimers() {
+                        clearTimeout(userInactivityLogoutTimer);
+                        userInactivityLogoutTimer = setTimeout(trySignoutRedirect, config.timeForUserInactivityAutomaticLogout);
+
+                        if (config.timeForUserInactivityNotification > 0) {
```

Comment:
if think the timeout should always be cleared if timers !== null. Then check the config to see if we need to re-enable.